### ISSUE TITLE
Weights are never passed for cross entropy in binary classification

### ISSUE THREAD

**Issue by ravinkohli** [automl/Auto-PyTorch#92]
Currently, to assign weights to the loss function, we are checking the output type, and if its binary we are making a kwargs dict with 'pos_weight': weights. This assumes that for binary classification the loss function is BCEWithLogitsLoss. However, that is not the case as by default we are using CrossEntropyLoss for binary classification and therefore the weights are never passed to this loss function.
